hack/validation/requirements.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# requirements validation for nodeclaim and nodepool
+# checking for restricted labels while filtering out well known labels
+
+rule=$'self in
+    [
+        "karpenter.azure.com/sku-name",
+        "karpenter.azure.com/sku-family",
+        "karpenter.azure.com/sku-version",
+        "karpenter.azure.com/sku-cpu",
+        "karpenter.azure.com/sku-memory",
+        "karpenter.azure.com/sku-accelerator",
+        "karpenter.azure.com/sku-networking-accelerated",
+        "karpenter.azure.com/sku-storage-premium-capable",
+        "karpenter.azure.com/sku-storage-ephemeralos-maxsize",
+        "karpenter.azure.com/sku-encryptionathost-capable",
+        "karpenter.azure.com/sku-gpu-name",
+        "karpenter.azure.com/sku-gpu-manufacturer",
+        "karpenter.azure.com/sku-gpu-count"
+    ]
+    || !self.find("^([^/]+)").endsWith("karpenter.azure.com")
```

Comment:
> !self.find("^([^/]+)"

Can we leave a comment explaining this regex a bit? It is a bit magical to me. Rule applies for any well known labels and any labels that do not end with "karpenter.azure.com"? 


The () creates a grouping for extracting a string. then the slash means we want to extract a substring using that `/` as a back reference?

Reply:
No, it just means "all non-slash characters after start" = "everything from start to the first /". I can add a comment ...